src/toil_scripts/lib/files.py

```diff
@@ -0,0 +1,78 @@
+from contextlib import closing
+import os
+import errno
+import tarfile
+import shutil
+
+
+# FIXME: replace with bd2k.util.files.mkdir_p
+def mkdir_p(path):
+    """
+    It is Easier to Ask for Forgiveness than Permission
+
+    :param str path: path to directory to create
+    """
+    try:
+        os.makedirs(path)
+    except OSError as exc:
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
+
+
+def tarball_files(tar_name, work_dir='.', prefix='', fpaths=None):
+    """
+    Creates a tarball from a group of files
+
+    :param str tar_name: Name of tarball
+    :param str work_dir: Current working directory
+    :param list[str] fpaths: File paths to include in the tarball
+    :param str prefix: Optional prefix for files in tarball
+    """
+    if fpaths is None:
+        fpaths = []
+    with tarfile.open(os.path.join(work_dir, tar_name), 'w:gz') as f_out:
+        for fpath in fpaths:
+            arcname = prefix + os.path.basename(fpath)
+            f_out.add(fpath, arcname=arcname)
+
+
+def move_files(filepaths, output_dir):
+    """
+    Moves files from the working directory to the output directory.
+
+    :param str output_dir: Output directory
+    :param list[str] filepaths: File paths to move
+    """
+    for fpath in filepaths:
+        dest = os.path.join(output_dir, os.path.basename(fpath))
+        shutil.move(fpath, dest)
+
+
+def consolidate_tarballs_job(job, **fname_to_id):
+    """
+    Combine the contents of separate tarballs into one.
+    Subdirs within the tarball will be named the keys in **fname_to_id
+
+    :param dict[str,str] fname_to_id: Key word arguments of the form: file_name_prefix=FileStoreID
+    :return str: FileStoreID
+    """
+    work_dir = job.fileStore.getLocalTempDir()
+    # Retrieve output file paths to consolidate
+    tar_paths = []
+    for fname, file_store_id in fname_to_id.iteritems():
+        p = job.fileStore.readGlobalFile(file_store_id, os.path.join(work_dir, fname + '.tar.gz'))
+        tar_paths.append((p, fname))
+    # I/O
+    output_name = 'foo.tar.gz'
```

Comment:
> It's a job function, so the name is irrelevant as it just returns a FileStoreId.

Ah, fair. Maybe drop a comment here to explain then. It makes sense upon explanation, but it wasn't obvious to me on the first pass of reading it, so I imagine that someone else will read over this code later and be confused.